ops/model.py

```diff
@@ -655,7 +655,10 @@ class NetworkInterface:
     def __init__(self, name: str, address_info: dict):
         self.name = name
         # TODO: expose a hardware address here, see LP: #1864070.
-        address = address_info.get('value')
+
+        # 'address' is the legacy key; 'value' is to-go from juju 2.9+
+        # see https://github.com/juju/juju/pull/14037
+        address = address_info.get('address') or address_info.get('value')
```

Comment:
I think you'd rather do it in the opposite order. If address is legacy it should only be read if 'value' isn't set.
```suggestion
        address = address_info.get('value') or address_info.get('address')
```
I often like the more explicit:
```suggestion
        address = address_info.get('value')
        if address is None: # Juju <2.9 only had an 'address' field
            address = address_info.get('address')
```